artifactory/commands/transferconfig/transferconfigmerge.go

```diff
@@ -0,0 +1,396 @@
+			if err != nil {
+				return
+			}
+			if diff != "" {
+				// Conflicts found, adding to conflicts CSV
+				*conflicts = append(*conflicts, Conflict{
+					Type:                Repository,
+					SourceName:          sourceRepo.Key,
+					TargetName:          sourceRepo.Key,
+					DifferentProperties: diff,
+				})
+			}
+		} else {
+			reposToTransfer = append(reposToTransfer, sourceRepo.Key)
+		}
+	}
+	if len(reposToTransfer) > 0 {
+		log.Info(fmt.Sprintf("Transferring %d repositories ...", len(reposToTransfer)))
+		err = tcmc.transferRepositoriesToTarget(reposToTransfer)
+	}
+	return
+}
+
+func (tcmc *TransferConfigMergeCommand) compareRepositories(sourceRepoBaseDetails, targetRepoBaseDetails services.RepositoryDetails) (diff string, err error) {
+	// Compare basic repository details
+	diff, err = compareInterfaces(sourceRepoBaseDetails, targetRepoBaseDetails, filteredRepoKeys...)
+	if err != nil && len(diff) != 0 {
+		return
+	}
+
+	// The basic details are equal. Continuing to compare the full repository details.
+	// Get full repo info from source and target
+	var sourceRepoFullDetails interface{}
+	err = tcmc.sourceArtifactoryManager.GetRepository(sourceRepoBaseDetails.Key, &sourceRepoFullDetails)
+	if err != nil {
+		return
+	}
+	var targetRepoFullDetails interface{}
+	err = tcmc.targetArtifactoryManager.GetRepository(targetRepoBaseDetails.Key, &targetRepoFullDetails)
+	if err != nil {
+		return
+	}
+	diff, err = compareInterfaces(sourceRepoFullDetails, targetRepoFullDetails, filteredRepoKeys...)
+	return
+}
+
+func compareInterfaces(first, second interface{}, filteredKeys ...string) (diff string, err error) {
+	diffList := []string{}
+	firstMap, err := coreutils.InterfaceToMap(first)
+	if err != nil {
+		return
+	}
+	secondMap, err := coreutils.InterfaceToMap(second)
+	if err != nil {
+		return
+	}
+	for key, firstValue := range firstMap {
+		if slices.Contains(filteredKeys, key) {
+			// Key should be filtered out
+			continue
+		}
+		if secondValue, ok := secondMap[key]; ok {
+			// Keys are only compared when exiting on both interfaces.
+			if !reflect.DeepEqual(firstValue, secondValue) {
+				diffList = append(diffList, key)
+			}
+		}
+	}
+	diff = strings.Join(diffList, "; ")
+	return
+}
+
+func (tcmc *TransferConfigMergeCommand) transferRepositoriesToTarget(reposToTransfer []string) (err error) {
+	// Decrypt source artifactory to get encrypted parameters
+	var wasEncrypted bool
+	if wasEncrypted, err = tcmc.sourceArtifactoryManager.DeactivateKeyEncryption(); err != nil {
+		return
+	}
+	defer func() {
+		if !wasEncrypted {
+			return
+		}
+		activationErr := tcmc.sourceArtifactoryManager.ActivateKeyEncryption()
+		if err == nil {
+			err = activationErr
+		}
+	}()
+	for _, repoKey := range reposToTransfer {
+		var params interface{}
+		err = tcmc.sourceArtifactoryManager.GetRepository(repoKey, &params)
+		if err != nil {
+			return
+		}
+		log.Info(fmt.Sprintf("Transferring '%s' configuration ...", repoKey))
```

Comment:
Transferring '%s' configuration ...
-->
Transferring the configuration of repository '%s'...